### PR TITLE
Bump version to 23.04

### PR DESCRIPTION
**Pull request by danielhollas**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by danielhollas** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {% set name = "aiidalab" %}
-{% set version = "23.04.0" %}
+{% set version = "23.03.0" %}
```
Comment (unkcpz) [inline recipe/meta.yaml]: Shame :grimacing:

**Review by unkcpz — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (danielhollas) [reply recipe/meta.yaml]: Nobody's perfect :smile:

**Comment by danielhollas** [comment]
@unkcpz FYI it looks like maintainers were not properly added last time in #3 because you ran the rerender before the commit that modified the maintainer list.

**Comment by unkcpz** [comment]
Seems still something not correct.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {% set name = "aiidalab" %}
-{% set version = "22.11.0" %}
+{% set version = "23.03.0" %}
```
Comment (unkcpz) [inline recipe/meta.yaml]:
```suggestion
{% set version = "23.3.0" %}
```

Need to be like this I guess

Comment (danielhollas) [reply recipe/meta.yaml]: Yes, turns out pypi somehow normalizes the version.

**Comment by danielhollas** [comment]
@conda-forge-admin, please rerender

**Comment by unkcpz** [comment]
The `sha256` problem, I think there must be some easy way to take care of this.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {% set name = "aiidalab" %}
-{% set version = "22.11.0" %}
+{% set version = "23.3.0" %}
 
 package:
   name: {{ name|lower }}
```
Comment (unkcpz) [inline recipe/meta.yaml:6]: See `sha256` below, need to be `4f4dd5f5635477ccc4a23c08a7ff03e3fbf206c3dc07f19338c5bedd94cda54a`

Comment (unkcpz) [reply recipe/meta.yaml:6]: You can move it to the head so we don't forget in the future, see https://github.com/conda-forge/plumpy-feedstock/blob/main/recipe/meta.yaml

Comment (danielhollas) [reply recipe/meta.yaml:6]: Good idea, done.

**Comment by unkcpz** [comment]
Finally :rocket:

**Pull request merged**